WIP: ci: frontend: add TestJob count numbers to build view

mwasilew opened the pull request:
TestJobs are counted by status. The number of each kind is displayed as
a badge in the build-nav.html. This patch also hides 'Test Jobs' tab
from the builds that don't include any test jobs.

Signed-off-by: Milosz Wasilewski <milosz.wasilewski@linaro.org>

mwasilew commented:
![testjob_badges](https://user-images.githubusercontent.com/618901/35157161-1002d0c4-fd2b-11e7-9657-a3ed65cea461.png)

The UI isn't complete yet but I wanted to get the opinion whether this is a good idea. Any comments are welcome.

terceiro commented on squad/frontend/templates/squad/build-nav.html:
```diff
@@ -18,11 +18,17 @@ <h1>
             All test results
         </a>
     </li>
+    {% if build.test_jobs.all %}
```
you might want to use `count` instead of `all` to avoid fetching the test jobs since you only want to know if there are any

mwasilew replied:
good point

terceiro commented on squad/frontend/templates/squad/build-nav.html:
```diff
@@ -18,11 +18,17 @@ <h1>
             Test jobs
         </a>
+        {% for i in build|filter_jobs %}
+        <span class="badge alert-success" data-toggle="tooltip" data-placement="top" title="Pass">{{ i.0 }} {{ i.1 }}</span>
```
won't this make the navigation tab too wide? say we had at least 3 different job status ...

mwasilew replied:
Yeah, I know. The UI might be difficult to do given I don't want to hardcode the names of the job statuses in the templates. I'll try to find some solution. This part is WIP.

danrue commented:
I tried looking at this but I couldn't get the tooltip to show up.. Is this the same data that shows up on the 'latest builds' page or different?

mwasilew commented:
It should not be a tooltip but the labels under "test jobs" tab. I guess it needs rebasing.

mwasilew commented:
![testjob_badges2](https://user-images.githubusercontent.com/618901/39469159-ca0c258a-4d2e-11e8-8d4a-90ec2cd52ab9.png)

Addressed comments. Maybe this is OK now?
